lib/Completion/Tests/Integration/Bridge/TolerantParser/WorseReflection/WorseClassMemberCompletorTest.php

```diff
@@ -503,6 +503,76 @@ class Foobar
             ],
         ];
 
+        yield 'Constant visibility from outside' => [
+            <<<'EOT'
+                <?php
+
+                class Foobar
+                {
+                    const FOOBAR = 'foobar';
+                    private const BARFOO = 'barfoo';
+                    protected const BARFOX = 'barfox';
+                }
+
+                $foobar = new Foobar();
+                $foobar::<>
+
+                EOT
+            , [
+                [
+                    'type' => Suggestion::TYPE_CONSTANT,
+                    'name' => 'FOOBAR',
+                    'short_description' => 'FOOBAR = "foobar"',
+                ],
+                [
+                    'type' => Suggestion::TYPE_CONSTANT,
+                    'name' => 'class',
+                    'short_description' => 'Foobar',
+                ],
+            ],
+        ];
+
+        yield 'Constant visibility from inside' => [
+            <<<'EOT'
+                <?php
+
+                class Foobar
+                {
+                    const FOOBAR = 'foobar';
+                    private const BARFOO = 'barfoo';
+                    protected const BARFOX = 'barfox';
+
+                    public function fog(): void
+                    {
+                        $foobar = self::<>
```

Comment:
I found that `static::<>` requires a separate fix - let create another issue for it.